Code/src/main/java/nl/utwente/group10/ui/components/blocks/function/ArgumentSpace.java

```diff
@@ -0,0 +1,373 @@
+            }
+            knot.setLayoutX(Math.min(Math.max(localX, leftBound), rightBound));
+            
+            // Properly react on the change in the knot's position.
+            double modifier = Math.max(-0.49, Math.min(KNOT_SNAP_MODIFIER, 0.49));
+            setKnotIndex((int) Math.round(determineKnotIndex() + modifier));
+            invalidateKnotPosition();
+        }
+    }
+    
+    /**
+     * @return Determines the index of the knot, based on the knot's current
+     *         position. This index has a decimal part, this represents the
+     *         percentage of the next argument that the knot has moved.
+     */
+    private double determineKnotIndex() {
+        if (Double.isNaN(getKnotPos())) {
+            //This is needed for when the knot does not have a position yet (initializing).
+            return getKnotIndex();
+        }
+        
+        double bowtieIndex = 0;
+        for (int i = 0; i < leftArguments.size(); i++) {
+            Region argument = leftArguments.get(i);
+            double min = argument.getLayoutX();
+            double max = min + argument.getWidth();
+            if (getKnotPos() > max) {
+                // Knot is positioned fully to the left of this argument.
+                bowtieIndex++;
+            } else {
+                // Knot is positioned somewhere halfway this argument.
+                bowtieIndex += Math.max(Math.min((getKnotPos() - min) / (max - min), 1), 0);
+                break;
+            }
+        }
+        return bowtieIndex;
+    }
+
+    /**
+     * This repositions the knot to a position based on the current knot index,
+     * regardless of the knot's current position.
+     * 
+     * Changes in Label length after snapping are accounted for by using
+     * property bindings.
+     */
+    public void snapToKnotIndex() {
+        if (inputID == ConnectionCreationManager.INPUT_ID_NONE) {
+            int kti = getKnotIndex();
+            if (kti > 0 && kti <= leftArguments.size()) {
+                Region arg = leftArguments.get(kti-1); // First argument left of the knot.
+    
+                // Binds the knot to the first argument on the left, so that if
+                // there is a length change to the left of the knot (because of a
+                // Label's text change), the knot moves accordingly.
+                knot.layoutXProperty().bind(arg.layoutXProperty().add(arg.widthProperty()).add(knot.radiusProperty()).add(H_GAP));
+            }else if (kti == 0) { // There is no argument left of the knot.
+                knot.setLayoutX(knot.getRadius());
+            }
+            
+            invalidateKnotPosition();
+        }
+    }
+    
+    /**
+     * Method to indicate that the position of the knot might have changed.
+     * 
+     * It repositions the arguments based on the current position of the knot.
+     */
+    public void invalidateKnotPosition() {
+        double bti = determineKnotIndex();
```

Comment:
bowTieIndex is quite long (yes, that would be very Java-like), but I see no problem here.